main.tf

```diff
@@ -181,3 +181,33 @@ resource "aws_wafv2_web_acl_association" "this" {
   resource_arn = aws_api_gateway_stage.this[count.index].arn
   web_acl_arn  = var.waf_acl
 }
+
+resource "aws_apigatewayv2_domain_name" "this" {
+  count = var.create_api_domain_name && length(var.stage_names) > 0 ? length(var.stage_names) : 0
+
+  domain_name = var.domain_names[count.index]
+
+  domain_name_configuration {
+    certificate_arn                        = var.domain_certificate_arn
+    ownership_verification_certificate_arn = var.domain_ownership_verification_certificate_arn
+    endpoint_type                          = "REGIONAL"
+    security_policy                        = "TLS_1_2"
+  }
+
+  dynamic "mutual_tls_authentication" {
+    for_each = length(keys(var.mutual_tls_authentication)) == 0 ? [] : [var.mutual_tls_authentication]
+
+    content {
+      truststore_uri     = mutual_tls_authentication.value.truststore_uri
+      truststore_version = try(mutual_tls_authentication.value.truststore_version, null)
+    }
+  }
+}
+
+resource "aws_apigatewayv2_api_mapping" "this" {
+  count = var.create_api_domain_name && length(var.stage_names) > 0 ? length(var.stage_names) : 0
+
+  api_id      = aws_api_gateway_rest_api.this.id
+  domain_name = aws_apigatewayv2_domain_name.this[count.index].id
+  stage       = aws_api_gateway_stage.this[count.index].stage_name
+}
```

Comment:
I'd like to know more about this approach using the api_gateway_domain_name & mapping (v2) to do this instead of the v1 resources available for this.

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_domain_name

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/api_gateway_base_path_mapping

Reply:
both works regardless the version and type of the API, imho v2 gets a bit better support for domain names than v1, but I can change it to use v1 only. Is strict requirement?

Reply:
It seems v2 doesn't support both EDGE and REGIONAL which is used in v1...

Reply:
moved to use v1 resources, thanks!